node/node.go

```diff
@@ -171,6 +173,24 @@ func NewNode(config *cfg.Config,
 	// reload the state (it may have been updated by the handshake)
 	state = sm.LoadState(stateDB)
 
+	// Connect to external signing process, if an address is provided.
+	if config.PrivValidatorAddr != "" {
+		var (
+			privKey = crypto.GenPrivKeyEd25519()
+			pvsc    = priv_val.NewSocketClient(
+				logger.With("module", "priv_val"),
+				config.PrivValidatorAddr,
+				&privKey,
+			)
+		)
+
+		if err := pvsc.Start(); err != nil {
```

Comment:
If we `defer pvsc.Stop()` here it would be killed as soon as we exit the function scope. If the client is used it's vital to the node and is kind of bound to the lifecycle of the node process itself, maybe I missed a place where we have teardown when the node shuts down, that would be the place to stop it.

Reply:
https://github.com/tendermint/tendermint/blob/master/node/node.go#L406